### ISSUE TITLE
Global/before all middleware

### ISSUE THREAD

**Issue by gtjamesa** [rawnly/next-wayfinder#6]
Hey,

I absolutely love the direction of this package, you've really improved on the default Next.js middleware experience. Unless I'm missing it entirely, this package lacks the ability to attach a global middleware that would otherwise be possible in the default Next.js middleware using [conditional statements](https://nextjs.org/docs/app/building-your-application/routing/middleware#conditional-statements).

Here is how I can somewhat see it working:

- Global middleware defined by the path `"*"` or via the options property such as `beforeAll` or `global`
- Can be typed using the existing `Middleware` type to allow the developer to skip certain paths (i.e. `_next`)
- Create a `response` object using `NextResponse.next()`
- If defined, MUST return a `NextResponse` object
- The returned response can then be supplied to each middleware `handler`:
    ```js
    {
      path: '/auth/:path*',
      handler: async (req, res) => {
        res.cookies.set('test2', 'test2');
    
        return res;
      },
    },
    ```

p.s. You should update the README on npm as it is blank at the moment

**Comment by rawnly** [comment]
Nice idea! I'll work on this asap! Thanks for the kind words I'm happy to know that it can be useful to others too :)

**Comment by rawnly** [comment]
In the meantime what you can do is:

```ts
export default function middleware(req, ev) {

  // < global middleware logic below
  const res = NextResponse.next();
  res.cookies.set('a', 'b')
  // >

  return handlePaths([...], { response: res })
}
```

I know this hasn't been documented well but it's a solution :)
i'll add the option to apply a global middleware anyway

**Comment by gtjamesa** [comment]
That's actually the implementation I ended up using right after submitting the issue. I've just seen the PR come through so I'll check it out when I get back home - wasn't expecting it so quickly! 😁

**Comment by rawnly** [comment]
Awesome feedback on PR is appreciated!

**Comment by gtjamesa** [comment]
I've tested it and it works perfectly, thank you. You've done an excellent job!

**Comment by rawnly** [comment]
Just released `0.2.3`!

https://www.npmjs.com/package/next-wayfinder
